### PR TITLE
chapter_3

### PR DESCRIPTION
**Pull request by balqis277**
ready

### CODE REVIEW

**Comment by balqis277** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -13,3 +13,7 @@
  *= require_tree .
  *= require_self
  */
+
+ @import 'bootstrap-sprockets';
+ @import 'bootstrap';
+
```
Comment (anlvq-1058) [inline app/assets/stylesheets/application.scss:19]: this line have space, delete all the space in last line

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <nav>
+     <ul class="nav navbar-nav navbar-right">
+     <li><%= link_to "Home", "#" %></li>
+     <li><%= link_to "Help", "#" %></li>
+     <li><%= link_to "Log in", "#" %></li>
+     </ul>
+    </nav>
+   </div>
+  </header>
+}
```
Comment (anlvq-1058) [inline app/views/layouts/_header.html.erb:13]:
```suggestion

```

File: app/views/static_pages/help.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+<h1>Help</h1>
+<p>
+ Get help on the Ruby on Rails Tutorial at the
+ <a href="http://www.railstutorial.org/help">Rails Tutorial help page</a>.
+ To get help on this sample app, see the
+ <a href="http://www.railstutorial.org/book"><em>Ruby on Rails Tutorial</em>
+ book</a>.
+</p>
+
```
Comment (anlvq-1058) [inline app/views/static_pages/help.html.erb:11]: delete the blank line

File: test/controllers/static_pages_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (anlvq-1058) [inline test/controllers/static_pages_controller_test.rb:1]: delete this file please

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    letter-spacing: -1px;
+    padding-top: 9px;
+    font-weight: bold;
+  }
+  
+  #logo:hover {
+    color: #fff;
+    text-decoration: none;
+  }
+
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss:42]: remove all spaces in line at the end of file

File: app/controllers/static_pages_controller.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
```
Comment (huongnt-2545) [inline app/controllers/static_pages_controller.rb:2]: if method hasn't no handle in
```
# Good
def home; end

def help; end
```

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb:2]: use the same indentation in all of files, vd: indent = 2 spaces

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+   <div class="container">
+    <%= link_to "sample app" , "#", id: "logo" %>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb:3]: indent = 2 spaces

File: app/views/layouts/_header.html.erb (hunk 2 above)
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb:13]: remove it `}`

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -11,6 +11,10 @@
   </head>
 
   <body>
-    <%= yield %>
+  <%= render "layouts/header" %>
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb:14]: indent

**Review by huongnt-2545 — COMMENTED** [review]
You have some comments not fix

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Comment by ngocvt-0484** [comment]
commented

**Comment by ngocvt-0484** [comment]
not fix comments yet

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Review by thaodtd-0665 — COMMENTED** [review]
This pull still have more comment, you need to fix this in next pull

**Comment by thaodtd-0665** [comment]
merged

**Pull request merged**